Rect42.dart

```diff
@@ -0,0 +1,121 @@
+import 'dart:math';
+
+import 'package:flutter/material.dart';
+
+class Rect42 extends StatefulWidget {
+  final int count;
+
+  const Rect42({Key? key, required this.count}) : super(key: key);
+
+  @override
+  State<Rect42> createState() => _Rect42State();
+}
+
+class ListColour<listRow> {
+  // class created to generate a radom colour for each row within the the ListView
+  late Color tileColour =
+      Colors.primaries[Random().nextInt(Colors.primaries.length)];
+
+  void changeColour1(tileColour) {
+    //method creatd to change the colour of the row
+    this.tileColour = tileColour;
+    tileColour = Colors.primaries[Random().nextInt(Colors.primaries.length)];
+  }
+}
+
+class _Rect42State extends State<Rect42> {
+  late List tiles;
+
+  @override
+  void initState() {
+    super.initState();
+    tiles = List.generate(widget.count, (index) => ListColour().tileColour);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: ListView.builder(
+        itemBuilder: (context, index) => buildRow(index),
+        itemCount: tiles.length,
+      ),
+    );
+  }
+
+  Widget buildRow(int index) {
+    final track = tiles[index];
+    final containerRow = Dismissible(
+      key: UniqueKey(),
+      background: Container(
+        color: Colors.red,
+        child: Row(
+          // ignore: prefer_const_literals_to_create_immutables
+          children: [
+            const Text('change colour'),
+            const Spacer(),
+            const Text('change colour'),
+          ],
+        ),
+      ),
+      child: Container(
+        width: MediaQuery.of(context).size.width,
+        height: 60,
+        color: track,
+      ),
+      onDismissed: (direction) {
+        setState(

```

Comment:
want you to use bloc

Reply:
I have moved this logic to a Bloc I created in the second pull request I uploaded.